pkg/apis/query/v0alpha1/register.go

```diff
@@ -0,0 +1,31 @@
+package v0alpha1
+
+import (
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+
+	common "github.com/grafana/grafana/pkg/apis/common/v0alpha1"
+)
+
+const (
+	GROUP      = "query.grafana.app"
+	VERSION    = "v0alpha1"
+	APIVERSION = GROUP + "/" + VERSION
+)
+
+var DataSourceResourceInfo = common.NewResourceInfo(GROUP, VERSION,
+	"datasources", "datasource", "DataSource",
+	func() runtime.Object { return &DataSource{} },
+	func() runtime.Object { return &DataSourceList{} },
+)
+
+var DataSourcePluginResourceInfo = common.NewResourceInfo(GROUP, VERSION,
```

Comment:
not needed now -- but I think this approach solves the desire need to have a single place to list datasources across all types.  This could be the API we use when listing what datasources exist for a tenant/user/token.

NOTE: we currently do not have an API that tells you your instances, it is rendered into the startup HTML and also available from `/api/frontend/settings#datasources`